Devscord.DiscordFramework/Middlewares/Factories/ReactionContextFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using Discord;
+using Discord.WebSocket;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+
+namespace Devscord.DiscordFramework.Middlewares.Factories
+{
+    internal class ReactionContextFactory : IContextFactory<Tuple<SocketReaction, IUserMessage>, ReactionContext>
+    {
+        public ReactionContext Create(Tuple<SocketReaction, IUserMessage> reactionAndUserMessage)
+        {
+            var reaction = reactionAndUserMessage.Item1;
+            var userMessage = reactionAndUserMessage.Item2;
+
+            var channelContext = new ChannelContextFactory().Create(reaction.Channel);
```

Comment:
te klasy typu ChannelContextFactory powinny iść z IoC

Reply:
a Factories są automatycznie rejestrowane w Autofac? Bo jeśli nie to lepiej jakbym po prostu w konstruktorze je normalnie zrobił.

Reply:
z tym że jeśli coś ma generyczny interfejs to musisz rejestrować to refleksjami wyciągając wszystkie obiekty
to ci nie przejdzie automatycznie